src/txn/txn.c

```diff
@@ -541,18 +541,20 @@ __wt_txn_reconfigure(WT_SESSION_IMPL *session, const char *config)
  * __wt_txn_release --
  *	Release the resources associated with the current transaction.
  */
-void
+int
 __wt_txn_release(WT_SESSION_IMPL *session)
 {
 	WT_TXN *txn;
 	WT_TXN_GLOBAL *txn_global;
 	WT_TXN_STATE *txn_state;
+	bool readonly;
 
 	txn = &session->txn;
 	txn_global = &S2C(session)->txn_global;
 	txn_state = WT_SESSION_TXN_STATE(session);
 
-	WT_ASSERT(session, txn->mod_count == 0);
+	readonly = txn->mod_count != 0;
```

Comment:
Is this backwards?  Wouldn't a readonly txn have a `txn->mod_count == 0`? If it is correct, it needs a comment because it is not intuitive.